### ISSUE TITLE
Version 1.23 doesn't work under Linux

### ISSUE THREAD

**Issue by zibik** [yarl/vicuna#86]
`java -jar vicuna-1.23/vicuna.jar 
maj 28, 2016 10:38:38 AM Wiki namespace
INFO: Fetching URL https://commons.wikimedia.org/w/api.php?maxlag=5&format=xml&action=query&meta=siteinfo&siprop=namespaces
maj 28, 2016 10:38:39 AM Wiki namespace
INFO: [commons.wikimedia.org] Successfully retrieved namespace list (39 namespaces)
maj 28, 2016 10:38:39 AM Wiki getPageText
INFO: Fetching URL https://commons.wikimedia.org/w/index.php?maxlag=5&title=User%3AYarl%2FVicunaUploader%2Fversion&action=raw
maj 28, 2016 10:38:39 AM Wiki getPageText
INFO: [commons.wikimedia.org] Successfully retrieved text of User:Yarl/VicunaUploader/version
Exception in thread "FIntro: Splash screen/Updater" com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter$UnknownFieldException: No such field cuploader.ImmutableCoordinate.org.openstreetmap.gui.jmapviewer.Coordinate
---- Debugging information ----
field               : org.openstreetmap.gui.jmapviewer.Coordinate
class               : cuploader.ImmutableCoordinate
required-type       : cuploader.ImmutableCoordinate
converter-type      : com.thoughtworks.xstream.converters.reflection.ReflectionConverter
path                : /settings/coor/org.openstreetmap.gui.jmapviewer.Coordinate
class[1]            : cuploader.Settings
version             : null

```
    at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.determineType(AbstractReflectionConverter.java:453)
    at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.doUnmarshal(AbstractReflectionConverter.java:294)
    at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.unmarshal(AbstractReflectionConverter.java:234)
    at com.thoughtworks.xstream.core.TreeUnmarshaller.convert(TreeUnmarshaller.java:72)
    at com.thoughtworks.xstream.core.AbstractReferenceUnmarshaller.convert(AbstractReferenceUnmarshaller.java:65)
    at com.thoughtworks.xstream.core.TreeUnmarshaller.convertAnother(TreeUnmarshaller.java:66)
    at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.unmarshallField(AbstractReflectionConverter.java:355)
    at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.doUnmarshal(AbstractReflectionConverter.java:306)
    at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.unmarshal(AbstractReflectionConverter.java:234)
    at com.thoughtworks.xstream.core.TreeUnmarshaller.convert(TreeUnmarshaller.java:72)
    at com.thoughtworks.xstream.core.AbstractReferenceUnmarshaller.convert(AbstractReferenceUnmarshaller.java:65)
    at com.thoughtworks.xstream.core.TreeUnmarshaller.convertAnother(TreeUnmarshaller.java:66)
    at com.thoughtworks.xstream.core.TreeUnmarshaller.convertAnother(TreeUnmarshaller.java:50)
    at com.thoughtworks.xstream.core.TreeUnmarshaller.start(TreeUnmarshaller.java:134)
    at com.thoughtworks.xstream.core.AbstractTreeMarshallingStrategy.unmarshal(AbstractTreeMarshallingStrategy.java:32)
    at com.thoughtworks.xstream.XStream.unmarshal(XStream.java:1058)
    at com.thoughtworks.xstream.XStream.unmarshal(XStream.java:1042)
    at com.thoughtworks.xstream.XStream.fromXML(XStream.java:913)
    at com.thoughtworks.xstream.XStream.fromXML(XStream.java:904)
    at cuploader.frames.Main.readSettings(Main.java:1153)                                                                                                                                 
    at cuploader.frames.Main.<init>(Main.java:56)                                                                                                                                         
    at cuploader.frames.FIntro$1.run(FIntro.java:41)                                                                                                                                      
    at java.lang.Thread.run(Thread.java:745)                                                                                                                                    `
```

**Comment by saper** [comment]
@Zibik how does your configuration file look like?

**Comment by mathieumd** [comment]
@zibik, v1.23 do work well on my Debian with OpenJDK  8. (But it does crash with OpenJDK 9, cf. #85)

You may try to install the package **jmapviewer**, maybe.

**Comment by ElGrafo** [comment]
Same problem for me on Linux Mint 17.

`$ java -version
java version "1.7.0_101"
OpenJDK Runtime Environment (IcedTea 2.6.6) (7u101-2.6.6-0ubuntu0.14.04.1)
OpenJDK 64-Bit Server VM (build 24.95-b01, mixed mode)`

jmapviewer is installed.

I noticed I had 2 files called "settings.vicuna": one in ~/vicuna/ (where the vicuna.jar is located) and one in ~/ The first one might be something I put there when I moved to another PC, removing it didn'T do anything. However, re-naming the second one to something like "settings.vicuna.old" and thus forcing vicuna to create a new one made vicuna start again.  So if you don't mind losing your setting, just delete the "settings.vicuna" in your home directory.

After experimenting a bit with my old settings file it seems like removing the following lines worls as well:
```
<coor serialization="custom">
    <org.openstreetmap.gui.jmapviewer.Coordinate>
      <double>11.578114628791809</double>
      <double>49.9441572560159</double>
    </org.openstreetmap.gui.jmapviewer.Coordinate>
  </coor>
  <coorZoom>18</coorZoom>
```

**Comment by nemobis** [comment]
With
```
$ java -version
openjdk version "1.8.0_212"
OpenJDK Runtime Environment (build 1.8.0_212-b04)
OpenJDK 64-Bit Server VM (build 25.212-b04, mixed mode)
```
and `jmapviewer-1.03-8` I also get:

```
$ java -jar vicuna.jar
Exception in thread "AWT-EventQueue-0" java.awt.HeadlessException
        at java.awt.GraphicsEnvironment.checkHeadless(GraphicsEnvironment.java:204)
        at java.awt.Window.<init>(Window.java:536)
        at java.awt.Frame.<init>(Frame.java:420)
        at java.awt.Frame.<init>(Frame.java:385)
        at javax.swing.JFrame.<init>(JFrame.java:189)
        at cuploader.frames.FIntro.<init>(FIntro.java:22)
        at cuploader.frames.FIntro$5.run(FIntro.java:247)
        at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
        at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
        at java.awt.EventQueue.access$500(EventQueue.java:97)
        at java.awt.EventQueue$3.run(EventQueue.java:709)
        at java.awt.EventQueue$3.run(EventQueue.java:703)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
        at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
        at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
        at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
        at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
        at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
        at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
        at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

No vicuna.settings file is found anywhere.

**Comment by nemobis** [comment]
Fixed by re-installing `java-1.8.0-openjdk`, while `dnf install java-openjdk-devel openjfx jmapviewer` may have been overkill.

**Comment by michal-josef-spacek** [comment]
I am closing this ticket. I don't want to support 1.23. Actually we have 1.24-beta2 and i could help with it.
